File: githubspace/IntelliJMac/Kotlin_Teacher/Chapter8/src/main/java/cn/kotliner/java/io/IO.java
package cn.kotliner.java.io;

import java.io.*;

/**
 * Created by benny on 5/28/17.
 */
public class IO {
    public static void main(String... args) {
        BufferedReader bufferedReader = null;
        try {
             bufferedReader = new BufferedReader(new FileReader(new File("build.gradle")));
            String line;
            while((line = bufferedReader.readLine()) != null){
                System.out.println(line);
            }
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            try {
                bufferedReader.close();
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }
}

File: githubspace/IntelliJMac/Kotlin_Teacher/Chapter8/src/main/kotlin/cn/kotliner/kotlin/io/IO.kt
package cn.kotliner.kotlin.io

import java.io.*

/**
 * Created by benny on 5/28/17.
 */
fun main(args: Array<String>) {
    val file = File("build.gradle")
    val bufferedReader = BufferedReader(FileReader(file))
    var line: String

    while(true){
        line = bufferedReader.readLine()?:break
        println(line)
    }
    bufferedReader.close()
}